src/Button/ButtonBase.tsx

```diff
@@ -6,26 +6,37 @@ import {useTheme} from '../ThemeProvider'
 import {ButtonProps, StyledButton} from './types'
 import {getVariantStyles, getSizeStyles, getButtonStyles} from './styles'
 
+const defaultSxProp = {}
+const iconWrapStyles = {
+  display: 'inline-block'
+}
+const trailingIconStyles = {
+  ...iconWrapStyles,
+  ml: 2
+}
+
 const ButtonBase = forwardRef<HTMLElement, ButtonProps>(
-  ({children, as: Component = 'button', sx: sxProp = {}, ...props}, forwardedRef): JSX.Element => {
+  ({children, as: Component = 'button', sx: sxProp = defaultSxProp, ...props}, forwardedRef): JSX.Element => {
     const {leadingIcon: LeadingIcon, trailingIcon: TrailingIcon, variant = 'default', size = 'medium'} = props
     const {theme} = useTheme()
-    const iconWrapStyles = {
-      display: 'inline-block'
-    }
     const sxStyles = useMemo(() => {
-      return merge.all([getButtonStyles(theme), getSizeStyles(size, variant, false), getVariantStyles(variant, theme)])
-    }, [theme, size, variant])
+      return merge.all([
+        getButtonStyles(theme),
+        getSizeStyles(size, variant, false),
+        getVariantStyles(variant, theme),
+        sxProp as SxProp
```

Comment:
I wonder if splitting these into 2 separate useMemos would be slightly better

I expect that the first 3 calls are _mostly static_ while the sxProp, if user provided, is unlikely to be memoized.

I'm not sure, but merging the first 3 once, and the user provided sx with that once, might be a bit better?

Reply:
```
const sxStyles = useMemo(() => merge.all([
        getButtonStyles(theme),
        getSizeStyles(size, variant, false),
        getVariantStyles(variant, theme)
]), [...])

const sxValue = useMemo(() => merge(sxStyles, sxProps), [sxStyles, sxProps])
```

it's _likely_ the case that when users provide sxProps, they'll often bail out of memoization here, but when they don't we should avoid re-merging these.

Reply:
Totally agreed, will split them up in this way 👍